app/jobs/prepare_move_notifications_job.rb

```diff
@@ -4,22 +4,25 @@
 class PrepareMoveNotificationsJob < ApplicationJob
   queue_as :notifications
 
-  def perform(topic_id:, action_name:)
+  def perform(topic_id:, action_name:, send_webhooks: true, send_emails: true, only_supplier_id: nil)
     move = Move.find(topic_id)
 
-    move.suppliers.each do |supplier|
+    # if the move has a specified supplier, use it; otherwise use the move.suppliers delegate based on from_location
+    [move.supplier || move.suppliers].flatten.each do |supplier|
```

Comment:
As far as I know, every move has a supplier attached to it, currently. So this could be simplified I think?

Reply:
I'm afraid not: there are currently 21 moves in production which do not have a `supplier_id`

Reply:
So that's interesting. I think those moves were created before the supplier FELTHAM A (HMPYOI) had the location/supplier yaml link applied.

Reply:
If we had a validation on moves to prevent a nil supplier_id in place then this could be simplified... but there might be other unexpected consequences, so best to tackle that problem separately

Reply:
Sounds good